src/components/Sidebars/TabSidebar.tsx

```diff
@@ -0,0 +1,130 @@
+    event.dataTransfer.setData('tabId', tabId)
+  }
+
+  const onDrop: DragEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault()
+    const draggedTabId = event.dataTransfer.getData('tabId')
+
+    let target = event.target as HTMLElement | null
+    // Iterate each child until we find the one we moved to
+    while (target && !target.getAttribute('data-tabid')) {
+      target = target.parentElement as HTMLElement | null
+    }
+
+    const targetTabId = target ? target.getAttribute('data-tabid') : null
+
+    if (draggedTabId && targetTabId) {
+      const draggedIndex = openTabs.findIndex((tab) => tab.id === draggedTabId)
+      const targetIndex = openTabs.findIndex((tab) => tab.id === targetTabId)
+      updateTabOrder(draggedIndex, targetIndex)
+    }
+  }
+
+  const onDragOver: DragEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault()
+  }
+
+  const handleMouseEnter = (e: React.MouseEvent<HTMLDivElement, MouseEvent>, tab: Tab) => {
+    const rect = e.currentTarget.getBoundingClientRect() || null
+    setHoveredTab(tab.filePath)
+    setTooltipPosition({
+      x: rect.left - 75,
+      y: rect.bottom - 5,
+    })
+  }
+
+  const handleMouseLevel = () => {
+    setHoveredTab(null)
+  }
+
+  return (
+    <div className="relative flex items-center whitespace-nowrap">
+      {openTabs.map((tab) => (
+        <div
+          id="titleBarSingleTab"
+          key={tab.id}
+          className="flex h-[10px] items-center justify-center"
+          onMouseEnter={(e) => handleMouseEnter(e, tab)}
+          onMouseLeave={handleMouseLevel}
+        >
+          <div
+            data-tabid={tab.id}
+            draggable="true"
+            onDragStart={(event) => onDragStart(event, tab.id)}
+            onDrop={onDrop}
+            onDragOver={onDragOver}
+            className={`relative flex w-[150px] cursor-pointer items-center justify-between gap-1 p-2 text-sm text-white
+              ${currentFilePath === tab.filePath ? 'rounded-md bg-dark-gray-c-three' : 'rounded-md'}`}
+            onClick={() => onTabSelect(tab)}
+          >
+            <span className="truncate">{removeFileExtension(tab.title)}</span>
+            <span
+              className="cursor-pointer px-1 hover:rounded-md hover:bg-dark-gray-c-five"
+              onClick={(e) => {
+                e.stopPropagation() // Prevent triggering onClick of parent div
+                onTabClose(e, tab.id)
+              }}
+            >
+              &times;
+            </span>
+            {hoveredTab === tab.filePath && <Tooltip filepath={tab.filePath} position={tooltipPosition} />}
+          </div>
+        </div>
+      ))}
+      {openTabs.length > 0 && (
+        <div
+          id="titleBarFileNavigator"
+          className="ml-1 flex h-[28px] cursor-pointer items-center justify-center px-2 text-white hover:rounded-md hover:bg-dark-gray-c-three"
+          onClick={() => {
+            window.electronUtils.showCreateFileModal('')
```

Comment:
Again as above - this functionality should be passed in either as a prop or a provider so that we avoid unnecessary extra calls to the backend

Reply:
I'll consider the provider, but I disagree with the prop suggestion. In reality, this won't be called frequently enough to worry about backend costs. However, making it a prop would make the code messier and less readable.